GVFS/GVFS.Common/HealthCalculator/EnlistmentPathData.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GVFS.Common
+{
+    public class EnlistmentPathData
+    {
+        public List<string> GitFolderPaths;
+        public List<string> GitFilePaths;
+        public List<string> PlaceholderFolderPaths;
+        public List<string> PlaceholderFilePaths;
+        public List<string> ModifiedFolderPaths;
+        public List<string> ModifiedFilePaths;
+
+        public EnlistmentPathData()
+        {
+            this.GitFolderPaths = new List<string>();
+            this.GitFilePaths = new List<string>();
+            this.PlaceholderFolderPaths = new List<string>();
+            this.PlaceholderFilePaths = new List<string>();
+            this.ModifiedFolderPaths = new List<string>();
+            this.ModifiedFilePaths = new List<string>();
+        }
+
+        public void AddGitTrackingPaths(IEnumerable<string> gitTrackingPaths)
+        {
+            // Ensure that both lists of paths have been normalized before unioning to ensure no duplicity
+            List<string> normalizedGitTrackingPaths = gitTrackingPaths.ToList();
+            this.NormalizePaths(normalizedGitTrackingPaths);

```

Comment:
Having multiple places where we normalize paths makes it hard to reason about when we are dealing with normalized vs un-normalized paths, and results in duplicate normalization calls for the same set of files.

Can we simplify the flow here? maybe by unioning the data after everything is normalized via the `NormalizePaths` method (maybe by stroring the paths with skip worktree bit set)?